walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java

```diff
@@ -22,6 +22,7 @@
    * Returns a greeting for the given name.
    */
   public String greet(String name) {
-    return "Hello " + name;
+    String modified = name.replaceAll("[@#$%]", "");
```

Comment:
Well, since you know regular expressions... Depending on the semantic interpretation of the word "like" in "symbols like @#$%", I could try to break it with a test like below!


```
  @Test
  public void testGreetingRemovesSymbols_ElenaBeingAnnoying() {
    Greeter greeter = new Greeter();

    String greeting = greeter.greet("   Ada^^_:  ");

    // symbols should be removed
    Assert.assertEquals("Hello Ada", greeting);
  }
```

1. Random side monologue: when designing security features, an include list (formerly known as "whitelist") is much better than an exclude list (formerly known as "blacklist").

2. Actually helpful comment: `[^abc]` in regex will match every character except `a`, `b` and `c`. And `[a-zA-Z]` will match any upper or lowercase English letter.